txmongo/connection.py

```diff
@@ -241,7 +241,7 @@ class ConnectionPool(object):
     __wc_possible_options = set(['w', "wtimeout", 'j', "fsync"])
 
     def __init__(self, uri="mongodb://127.0.0.1:27017", pool_size=1, ssl_context_factory=None, **kwargs):
-        assert isinstance(uri, str)
+        assert isinstance(uri, StringType)
```

Comment:
There is a convention to raise `TypeError` when arguments of invalid types are passed... But I agree that `assert` looks shorter and cleaner. Anyways, it is out of scope of this pull.